File: app/src/main/java/com/xephorium/crystalnote/ui/base/BasePresenter.java
package com.xephorium.crystalnote.ui.base;

public class BasePresenter<T extends BaseView> implements Presenter<T> {

    private T view;

    @Override
    public void attachView(T view) {
        this.view = view;
    }

    @Override
    public void detachView() {
        this.view = null;
    }

    public boolean isViewAttached() {
        return this.view != null;
    }

    public T getView() {
        return this.view;
    }
}

File: app/src/main/java/com/xephorium/crystalnote/ui/base/BasePresenter.kt
package com.xephorium.crystalnote.ui.base

open class BasePresenter<T : BaseView> : Presenter<T> {

    var view: T? = null
        private set

    val isViewAttached: Boolean
        get() = this.view != null

    override fun attachView(view: T) {
        this.view = view
    }

    override fun detachView() {
        this.view = null
    }
}
